src/vm/perfinfo.cpp

```diff
@@ -0,0 +1,125 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ===========================================================================
+// File: perfinfo.cpp
+//
+
+#include "common.h"
+
+#if defined(FEATURE_PERFMAP) && !defined(DACCESS_COMPILE)
+#include "perfinfo.h"
+#include "pal.h"
+
+PerfInfo::PerfInfo(int pid)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    WCHAR tempPath[MAX_LONGPATH+1];
+    if (!GetTempPathW(MAX_LONGPATH, tempPath))
```

Comment:
We have recently tried to erradicate all fixed size buffers from the codebase. Please use WszGetTempPath with SString parameter instead.